commit 02bcc8b6a2e7f0cd2cf8eeae445215df54c0f9ba (refs/changes/41/5441/4)
Author: Joshua M. Clulow <jmc@joyent.com>
Date:   2019-02-05T22:21:11+00:00 (8 months ago)
    
    MANTA-4099 Muskie test suite could make it easier to set up access control
    Reviewed by: Jan Wyszynski <jan.wyszynski@joyent.com>
    Reviewed by: Cody Peter Mello <melloc@writev.io>
    Approved by: Jan Wyszynski <jan.wyszynski@joyent.com>

diff --git a/Makefile b/Makefile
index 96f2cf3..35a266c 100644
--- a/Makefile
+++ b/Makefile
@@ -68,6 +68,18 @@ test: $(STAMP_NODE_MODULES)
 	    $(NODE) ./node_modules/.bin/nodeunit --reporter=tap \
 	    test/*.test.js test/mpu/*.test.js
 
+#
+# This target can be used to invoke "acsetup.js", a program which configures
+# access control in the current Manta account in preparation for running the
+# Muskie test suite.  The most common invocations will include:
+#
+#	make test-ac-setup
+#	make test-ac-teardown
+#
+.PHONY: test-ac-%
+test-ac-%: $(STAMP_NODE_MODULES)
+	PATH=$(ROOT)/$(NODE_INSTALL)/bin:$(PATH) $(NODE) test/acsetup.js $*
+
 .PHONY: release
 release: all docs
 	@echo "Building $(RELEASE_TARBALL)"
diff --git a/README.md b/README.md
index 147af10..58f95b8 100644
--- a/README.md
+++ b/README.md
@@ -118,7 +118,7 @@ To run the tests against changes in a repository:
    modules.
 2. Configure your user account for access control by running:
 
-        $ build/node/bin/node test/acsetup.js setup
+        $ make test-ac-setup
 
    This uses the local copy of Node (pulled down as part of the build) to run
    the access-control setup script.  This creates roles and other server-side
@@ -127,9 +127,7 @@ To run the tests against changes in a repository:
 
         $ build/node/bin/node main.js -f etc/config.json
 
-4. Run `make test`.  (Due to
-   [npm issue #4191](https://github.com/npm/npm/issues/4191), this step always
-   reinstalls several dependencies.)
+4. Run `make test`.
 
 If you're changing anything about the way muskie is deployed, configured, or
 started, you should definitely test creating a muskie image and deploying that
@@ -150,7 +148,8 @@ check first before reporting problems:
    then check to see that your `MANTA_USER` is indeed **not** an operator.
 2. If a test fails with an InvalidRoleTag error, whose message may say something
    like 'Role tag "muskie\_test\_role\_default" is invalid.', then check that
-   you ran the acsetup script described above for the user that you're using.
+   you ran `make test-ac-setup` as described above for the user that you're
+   using.
    (Note that you may see some other muskie\_test\_role in the message.)
 3. If a test fails with a message like "Error: MUSKIE\_SALT required", then
    check that you've specified the three `MUSKIE_` environment variables
